File: src/C#/Usuario.cs
using System;

class Usuario
{
    // Atributos
    private string nombre = "";
    private string apellidos = "";
    private int edad;
    

    // Constructor por defecto
    public Usuario()
    {
        this.nombre = "";
        this.apellidos = "";
        this.edad = 0;
    }

    // Constructor
    public Usuario(string nombre, string apellidos, int edad)
    {
        this.nombre = nombre;
        this.apellidos = apellidos;
        this.edad = edad;
    }

    // Setter y getter
    public String Nombre
    {
        get { return this.nombre; }
        set { this.nombre = value; }
    }

    public String Apellidos
    {
        get { return this.apellidos; }
        set { this.apellidos = value; }
    }

    public int Edad
    {
        get { return this.edad; }
        set { this.edad = value; }
    }

    // Métodos
    public void hacerReporte()
    {
        Console.WriteLine("El usuario " + this.nombre + " esta iniciando sesion.");
    }

    public void iniciarSesion() 
    {
        Console.WriteLine("El usuario " + this.nombre + " ha cerrado la sesion.");
    }

    public void cerrarSesion() 
    {
        Console.WriteLine("Reporte de usuario");
        Console.WriteLine("Nombre completo: " + this.nombre + " " + this.apellidos);
        Console.WriteLine("Edad: " + this.edad);
    }
}

File: src/Java/Usuario.java
package Java;

public class Usuario {

    // Atributos
    String nombre;
    String apellidos;
    int edad;

    // Constructor
    public Usuario() {
        this.nombre = "";
        this.apellidos = "";
        this.edad = 0;
    }

    // Sobrecarga de constructores
    public Usuario(String nombre, String apellidos, int edad) {
        this.nombre = nombre;
        this.apellidos = apellidos;
        this.edad = edad;
    }

    // Declaración de los setter y getter
    void setNombre(String nombre) {
        this.nombre = nombre;
    }

    String getNombre() {
        return this.nombre;
    }

    void setApellidos(String apellidos) {
        this.apellidos = apellidos;
    }

    String getApellidos() {
        return this.apellidos;
    }

    void setEdad(int edad) {
        this.edad = edad;
    }

    int getEdad() {
        return this.edad;
    }

    // Métodos
    public void iniciarSesion() {
        System.out.println("El usuario " + this.nombre + " esta iniciando sesion.");
    }

    public void cerrarSesion() {
        System.out.println("El usuario " + this.nombre + " ha cerrado la sesion.");
    }

    public void hacerReporte() {
        System.out.println("Reporte de usuario");
        System.out.println("Nombre completo: " + this.nombre + " " + this.apellidos);
        System.out.println("Edad: " + this.edad);
    }
}